File: examples/readhtml.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>

#include <libxml/HTMLparser.h>

htmlDocPtr gethtml(char *doclocation,char *encoding) {
    htmlDocPtr doc;

    doc = htmlReadFile(doclocation, encoding, HTML_PARSE_NOBLANKS | HTML_PARSE_NOERROR | HTML_PARSE_NOWARNING | HTML_PARSE_NONET);

    if (doc == NULL) {
        fprintf(stderr, "Document not parsed successfully.\n");
        return;
    }

    return doc;
}

void getroot(htmlDocPtr doc) {
    xmlNode *cur = NULL;

    cur = xmlDocGetRootElement(doc);

    if (cur == NULL) {
        fprintf(stderr, "empty document\n");
        xmlFreeDoc(doc);
        return;
    }

    printf("%s\n", cur->name);

}

int main(void) {
    char *website = "http://www.google.com/index.html";
    char *encoding = "UTF-8";
    htmlDocPtr doc;

    doc = gethtml(website, encoding);

    getroot(doc);

    return 0;
}

File: _examples/readhtml.go
package main

import (
	. "github.com/martende/goxml"
	"os"
	"fmt"
)

func GetHtml(doclocation string,encoding string) (doc *XmlDoc,err error) {    
    doc,err= HtmlReadFile(doclocation, encoding, HTML_PARSE_NOBLANKS | HTML_PARSE_NOERROR | HTML_PARSE_NOWARNING | HTML_PARSE_NONET)
    if err!=nil {
    	fmt.Printf( "Document Parsing Failed %s\n",err)
    }    	
    return
}

func GetRoot(doc *XmlDoc) {
    
    cur := XmlDocGetRootElement(doc)

    if (cur == nil) {
        fmt.Printf( "empty document\n")
    } else {
    	fmt.Printf("RootElement %s\n", cur.GetName())
    }

}


func main() {
	website:="http://xmlsoft.org/"
	encoding:="UTF-8"
	
	XmlCheckVersion()
	
	if len(os.Args) >= 2 {
		website = os.Args[1]
        
    }
    
    doc,err := GetHtml(website, encoding);
    if err != nil {
    	panic(err)
    }
    
    GetRoot(doc)
    XmlFreeDoc(doc)
	XmlCleanupParser()
	XmlMemoryDump()
}

